app/api/users.js

```diff
@@ -0,0 +1,76 @@
+// User creation and authentication with password hashing.
+// Code sourced from Zybooks 11.12 https://learn.zybooks.com/zybook/UMASSCOMPSCI326RichardsAcademicYear2020/chapter/11/section/12
+const jwt = require("jwt-simple");
+const User = require("../models/user");
+const router = require("express").Router();
+const bcrypt = require("bcrypt-nodejs");
+const saltRounds = "10";
+
+const secret = "secret";
+
+router.post("/user", (req, res) => {
+    bcrypt.hash(req.body.password, saltRounds, null, (err, hash) => {
+        let newUser = new User({
+            username: req.body.username,
+            password: hash,
+            status: req.body.status
+        });
+
+        newUser.save((err) => {
+            if (err) return next(err);
```

Comment:
This is how it is written in zybooks, not sure why they use next but there's probably a reason?

Reply:
lmao the code in Zybooks is invalid. If they had specified `next` as the third parameter in the middleware function signature (i.e. `router.post("/user", function(req, res, next) {`), calling `next()` would have passed the request to the next middleware function. Since they didn't specify `next` in the middleware function signature, it's undefined and won't work.

Since we're not chaining together middleware, we should replace the call to `next()` with code to handle the error immediately by doing something like `res.status(500).json({error: "Error creating user"})`

Reply:
That's stupid of them.